app/services/record_declarations/base.rb

```diff
@@ -61,33 +61,34 @@ def parsed_date
     end
 
     def create_declaration_attempt!
-      ParticipantDeclarationAttempt.create!(
-        course_identifier: course_identifier,
-        declaration_date: declaration_date,
-        declaration_type: declaration_type,
-        cpd_lead_provider: cpd_lead_provider,
-        user: user,
-      )
+      ParticipantDeclarationAttempt.create!(declaration_parameters)
     end
 
     def find_or_create_record!
-      ActiveRecord::Base.transaction do
-        self.class.declaration_model.find_or_create_by!(
-          course_identifier: course_identifier,
-          declaration_date: declaration_date,
-          declaration_type: declaration_type,
-          cpd_lead_provider: cpd_lead_provider,
-          user: user,
-        ) do |participant_declaration|
-          profile_declaration = ProfileDeclaration.create!(
+      ApplicationRecord.transaction do
+        self.class.declaration_model.find_or_create_by!(declaration_parameters) do |participant_declaration|
+          DeclarationState.create!(
+            participant_declaration: participant_declaration,
+            state: user_profile.fundable? ? "payable" : "submitted",
+          )
```

Comment:
So we sometimes start in one state and sometimes in another state? I guess it's alright.

Reply:
Yes. If it's deemed payable at the point of creation, then it's set that way. I actually want the eligibility checks to trigger this state change so we don't need to recheck it later.